File: backends/Contracts/Handler/IHandlerContext.cs
using Microsoft.Extensions.Logging;

namespace D2.Contracts.Handler;

/// <summary>
/// Defines a context for handlers, providing access to the request context and logging
/// capabilities.
/// </summary>
public interface IHandlerContext
{
    /// <summary>
    /// Gets the request context associated with the handler.
    /// </summary>
    IRequestContext Request { get; }

    /// <summary>
    /// Gets the logger for logging within the handler.
    /// </summary>
    ILogger Logger { get; }
}

File: backends/node/shared/handler/src/i-handler-context.ts
import type { ILogger } from "@d2/logging";
import type { IRequestContext } from "./i-request-context.js";

/**
 * Context for handlers, providing access to the request context and logging.
 * Mirrors D2.Shared.Handler.IHandlerContext in .NET.
 */
export interface IHandlerContext {
  request: IRequestContext;
  logger: ILogger;
}
